extensions/frontend-tiles/src/integration-test/TestContext.ts

```diff
@@ -0,0 +1,57 @@
+import { AccessToken } from "@itwin/core-bentley";
+import { getAccessTokenFromBackend } from "@itwin/oidc-signin-tool/lib/cjs/frontend";
+
+export class TestContext {
+  private _accessToken: AccessToken | undefined;
+  private static _instance?: TestContext = undefined;
+
+  private constructor() { }
+
+  public static async instance(): Promise<TestContext> {
+    if (this._instance === undefined) {
+      this._instance = new TestContext();
+      await this._instance.initialize();
+    }
+    return this._instance;
+  }
+
+  public getAccessToken(): AccessToken | undefined {
+    return this._accessToken;
+  }
+
+  private async initialize() {
+    const urlPrefix = process.env.IMJS_URL_PREFIX || "";
+    const oidcConfig = {
+      clientId: process.env.IMJS_MES_INTEGRATION_OIDC_CLIENT_ID || "",
```

Comment:
Just an optimization: might be worth it to check at the beginning of `initialize()` whether the env variables IMJS_MES_INTEGRATION_OIDC_CLIENT_ID, IMJS_TEST_REGULAR_USER_NAME, and IMJS_TEST_REGULAR_PASSWORD exist. Because if they don't exist, the test will definitely fail so there's no need for retries (doesn't apply to the url prefix, which can correctly default to "")